front50-s3/src/main/java/com/netflix/spinnaker/front50/config/S3ClientFactory.java

```diff
@@ -37,39 +37,46 @@
   public static AmazonS3 create(
       AWSCredentialsProvider awsCredentialsProvider, S3Properties s3Properties) {
     ClientConfiguration clientConfiguration = new ClientConfiguration();
+
     if (s3Properties.getProxyProtocol() != null) {
       if (s3Properties.getProxyProtocol().equalsIgnoreCase("HTTPS")) {
         clientConfiguration.setProtocol(Protocol.HTTPS);
       } else {
         clientConfiguration.setProtocol(Protocol.HTTP);
       }
+
       Optional.ofNullable(s3Properties.getProxyHost()).ifPresent(clientConfiguration::setProxyHost);
       Optional.ofNullable(s3Properties.getProxyPort())
           .map(Integer::parseInt)
           .ifPresent(clientConfiguration::setProxyPort);
     }
+
     if (!StringUtils.isEmpty(s3Properties.getSignerOverride())) {
       clientConfiguration.setSignerOverride(s3Properties.getSignerOverride());
     }
 
     AmazonS3Client client = new AmazonS3Client(awsCredentialsProvider, clientConfiguration);
 
+    S3ClientOptions.Builder clientOptionsBuilder =
+        S3ClientOptions.builder().setPayloadSigningEnabled(s3Properties.getPayloadSigning());
```

Comment:
Just to be extra safe it would be better to check if `s3Properties.getPayloadSigning()` is not null (or convert to `boolean` primitive) before setting this property, in order to avoid `NullPointerException` if the value is ever null.